src/py21cmfast/src/Stochasticity.c

```diff
@@ -860,15 +860,8 @@ int sample_halo_grids(gsl_rng **rng_arr, double redshift, float *dens_field,
                         if (hm_buf[i] < simulation_options_global->SAMPLER_MIN_MASS) continue;
 
                         if (count >= arraysize_local) {
-                            LOG_ERROR(
-                                "More than %llu halos (expected %.1e) with buffer size factor %.1f",
-                                arraysize_local,
-                                arraysize_local / config_settings.HALO_CATALOG_MEM_FACTOR,
-                                config_settings.HALO_CATALOG_MEM_FACTOR);
-                            LOG_ERROR(
-                                "If you expected to have an above average halo number try raising "
-                                "config_settings.HALO_CATALOG_MEM_FACTOR");
-                            Throw(ValueError);
+                            OUT_OF_BUFFER = true;
+                            continue;
```

Comment:
Since `count` is a private variable, the other threads will run through entirely. To avoid this you can set an `if(OUT_OF_BUFFER) continue;` at the start of the loop